src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/MarkupChunkGenerator.cs

```diff
@@ -7,6 +7,10 @@ namespace Microsoft.AspNetCore.Razor.Language.Legacy;
 
 internal class MarkupChunkGenerator : SpanChunkGenerator
 {
+    public static readonly MarkupChunkGenerator Instance = new();
```

Comment:
Did you consider a similar change for `StatementChunkGenerator`? It also seems like it can be a singelton.

Reply:
I didn't, mainly because there were less than 100 instances of chunk generators _other_ than markup generators, but I can do this.